src/PythonQt.cpp

```diff
@@ -502,11 +502,11 @@ void PythonQt::setRedirectStdInCallbackEnabled(bool enabled)
   sys.setNewRef(PyImport_ImportModule("sys"));
 
   if (enabled) {
-    if( !PyObject_HasAttrString(sys.object(), "pythonqt_stdin") ) {
+    if( PyObject_HasAttrString(sys.object(), "pythonqt_stdin") ) {
```

Comment:
I don't understand how this has worked before - also would ask @usiems to review, who was been involved here.

Reply:
The commit I reverted was about memory leaks (which caused crashes, thus the revert). I didn't realize that the commit also included another bug fix.